src/test/java/com/feedzai/commons/sql/abstraction/engine/impl/abs/EngineGeneralTest.java

```diff
@@ -1989,6 +1992,109 @@ public void betweenTest() throws DatabaseEngineException {
         );
     }
 
+    @Test
+    public void testWith() throws DatabaseEngineException {
+        if (this.engine instanceof MySqlEngine) {
+            // MySQL does not support With
+            return;
```

Comment:
Should we assert that if MySQL you get the not supported exception?

Reply:
I believe this makes sense, you should assert different things according to the engine.